google/cloud/spanner/client.h

```diff
@@ -429,79 +326,32 @@ class Client {
-    return ExecuteQuery(std::move(transaction_options), std::move(statement),
-                        Options(opts));
-  }
-  /**
-   * @copybrief ExecuteQuery(Transaction::SingleUseOptions,SqlStatement,Options)
-   * @see ExecuteQuery(Transaction::SingleUseOptions,SqlStatement,Options)
-   */
-  RowStream ExecuteQuery(Transaction::SingleUseOptions transaction_options,
-                         SqlStatement statement,
-                         std::initializer_list<internal::NonConstructible>) {
-    return ExecuteQuery(std::move(transaction_options), std::move(statement));
-  }
-  /**
-   * @copybrief ExecuteQuery(Transaction,SqlStatement,Options)
-   * @see ExecuteQuery(Transaction,SqlStatement,Options)
-   */
-  RowStream ExecuteQuery(Transaction transaction, SqlStatement statement,
-                         QueryOptions const& opts) {
-    return ExecuteQuery(std::move(transaction), std::move(statement),
-                        Options(opts));
-  }
-  /**
-   * @copybrief ExecuteQuery(Transaction,SqlStatement,Options)
-   * @see ExecuteQuery(Transaction,SqlStatement,Options)
-   */
-  RowStream ExecuteQuery(Transaction transaction, SqlStatement statement,
-                         std::initializer_list<internal::NonConstructible>) {
-    return ExecuteQuery(std::move(transaction), std::move(statement));
-  }
-  /**
-   * @copybrief ExecuteQuery(QueryPartition const&,Options)
-   * @see ExecuteQuery(QueryPartition const&,Options)
-   */
-  RowStream ExecuteQuery(QueryPartition const& partition,
-                         QueryOptions const& opts) {
-    return ExecuteQuery(partition, Options(opts));
-  }
   /**
-   * @copybrief ExecuteQuery(QueryPartition const&,Options)
-   * @see ExecuteQuery(QueryPartition const&,Options)
+   * Creates a set of partitions that can be used to execute a query
+   * operation in parallel.  Each of the returned partitions can be passed
+   * to `ExecuteQuery` to specify a subset of the query result to read.
+   *
+   * Partitions become invalid when the session used to create them is deleted,
+   * is idle for too long, begins a new transaction, or becomes too old. When
+   * any of these happen, it is not possible to resume the query, and the whole
+   * operation must be restarted from the beginning.
+   *
+   * @par Example
+   * @snippet samples.cc partition-query
+   *
+   * @param transaction The transaction to execute the operation in.
+   *     **Must** be a read-only snapshot transaction.
+   * @param statement The SQL statement to execute.
+   * @param opts `Options` used for this request.
+   *
+   * @return A `StatusOr` containing a vector of `QueryPartition`s or error
+   *     status on failure.
    */
-  RowStream ExecuteQuery(QueryPartition const& partition,
-                         std::initializer_list<internal::NonConstructible>) {
-    return ExecuteQuery(partition);
-  }
-  ///@}
+  StatusOr<std::vector<QueryPartition>> PartitionQuery(
```

Comment:
(FYI, you might have already noticed that) the ordering used to be:
1. `ExecuteQuery`
2. `ProfileQuery`
3. `PartitionQuery`

and the .cc file [uses that ordering](https://github.com/googleapis/google-cloud-cpp/blob/3ae5c1ce53466090327348f849d197aff838e851/google/cloud/spanner/client.cc#L131-L173)

I think the diff will be cleaner if you restore that order.

Reply:
> I think the diff will be cleaner if you restore that order.

Done.  I thought I might be improving things, but that's questionable, and it did certainly make it more confusing.  Thanks.